owncloudApp/src/main/java/com/owncloud/android/presentation/files/filelist/MainFileListFragment.kt

```diff
@@ -348,30 +348,37 @@ class MainFileListFragment : Fragment(),
     fun updateFileListOption(newFileListOption: FileListOption, file: OCFile) {
         mainFileListViewModel.updateFolderToDisplay(file)
         mainFileListViewModel.updateFileListOption(newFileListOption)
-        showOrHideFab(newFileListOption)
+        showOrHideFab(newFileListOption, file)
     }
 
     /**
-     * Check whether the fab should be shown or hidden depending on the [FileListOption]
+     * Check whether the fab should be shown or hidden depending on the [FileListOption] and
+     * the current folder displayed permissions
      *
      * Show FAB when [FileListOption.ALL_FILES] and not picking a folder
      * Hide FAB When [FileListOption.SHARED_BY_LINK], [FileListOption.AV_OFFLINE] or picking a folder
      *
      * @param newFileListOption new file list option to enable.
      */
-    private fun showOrHideFab(newFileListOption: FileListOption) {
-        if (!newFileListOption.isAllFiles() || isPickingAFolder()) {
+    private fun showOrHideFab(newFileListOption: FileListOption, currentFolder: OCFile) {
+        if (!newFileListOption.isAllFiles() || isPickingAFolder() || (currentFolder.permissions != null && !currentFolder.hasAddFilePermission && !currentFolder.hasAddSubdirectoriesPermission)) {
             toggleFabVisibility(false)
         } else {
             toggleFabVisibility(true)
-
+            if (currentFolder.permissions != null) {
+                if (!currentFolder.hasAddFilePermission) {
+                    binding.fabUpload.isVisible = false
+                } else if (!currentFolder.hasAddSubdirectoriesPermission) {
+                    binding.fabMkdir.isVisible = false
+                }
+            }
```

Comment:
This logic and the 
```
|| (currentFolder.permissions != null && !currentFolder.hasAddFilePermission && !currentFolder.hasAddSubdirectoriesPermission
```
looks like they are redundant and could be unified. At this point, we show the fab and then we hide it depending on permissions. And the logic I think it would be permissions and then check filelistoption only in case there are permsisions

Also, we do not need to register the fab listeners in case they are hidden.

Reply:
The alternative would be repeating the `toggleFabVisibility(false)` logic instead of the conditions logic, and also the main FAB would be made visible and then maybe invisible again if we go into the "else", so I think we have the same logic structure problem in both cases :S